pkg/controllers/podgroup/pg_controller_handler.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2019 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podgroup
+
+import (
+	"github.com/golang/glog"
+
+	"k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kbv1 "github.com/kubernetes-sigs/kube-batch/pkg/apis/scheduling/v1alpha1"
+
+	vkbatchv1 "volcano.sh/volcano/pkg/apis/batch/v1alpha1"
+	"volcano.sh/volcano/pkg/apis/helpers"
+)
+
+type podRequest struct {
+	pod *v1.Pod
+}
+
+func (cc *Controller) addPod(obj interface{}) {
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		glog.Errorf("Failed to convert %v to v1.Pod", obj)
+		return
+	}
+
+	req := podRequest{
+		pod: pod,
+	}
+
+	cc.queue.Add(req)
+}
+
+func needCreatePG(pod *v1.Pod) bool {
+	if pod == nil {
+		return false
+	}
+
+	isNormalPod := false
```

Comment:
```
	if pod.Annotations == nil || pod.Annotations[vkbatchv1.JobNameKey] == "" {
		if pod.Spec.SchedulerName != v1.DefaultSchedulerName {
			return true
		}
	}

```

And this is the logic which can be moved into the informer's filter function.

Reply:
delete the func , move if to podInformer AddFunc.